pkgs/tools/misc/resorter/resorter.r

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/Rscript
```

Comment:
Why such a huge vendoring inside Nixpkgs?

Reply:
We _could_ instead use [this fork](https://github.com/hiAndrewQuinn/resorter) which is uploaded to GitHub,
but I believe it's needlessly complex considering that the tool is unlikely to ever be updated, barring breaking changes to R.

Reply:
> We could instead use [this fork](https://github.com/hiAndrewQuinn/resorter) which is uploaded to GitHub,

Use it, then.

> but I believe it's needlessly complex considering that the tool is unlikely to ever be updated, 

We have pieces of software with decades of existence in Nixpkgs. We never vendored them inside Nixpkgs just because the upstream never changed them.

> barring breaking changes to R.

Well, you have given a huge reason to not vendor this script.

Not only one but two: a Nixpkgs contributor should not be obliged to have previous knowledge of R language in order to contribute with Nixpkgs. Ideally only Nix and rudimentary Bash, maybe some sed&awk, are required to contribute.
What if some subtle bug or optimization is discovered? Should we the Nixpkgs team update this software, becoming the maintainers of an R script?

A third reason: Nixpkgs is not the only package manager in this world.
What if some Archlinux user likes this script and wants to package it at AUR?
Should them be forced to download the whole Nixpkgs just to extract a single file from it?

It would be better to create a GitHub account to share it with everyone - [ops, it was done already](https://github.com/hiAndrewQuinn/resorter)!

> ~100 LOC is also really not that large compared to say an NPM or Yarn lockfile.

Lockfiles are different beasts.

Yarn lockfiles are machine-generated; indeed they are not even intended to be read by human beings.
We generally include them on Nixpkgs because it is easier to update them instead of regenerating from scratch.